docs/component_branding.md

```diff
@@ -0,0 +1,47 @@
+# Component branding
+
+## Overview
+
+Organisation pages in whitehall ([example](https://www.gov.uk/government/organisations/attorney-generals-office)) have department-specific branding on them. This includes specific colours for links and borders. These pages are being migrated out of whitehall and will use components for their frontend, which means some components need a sensible way of displaying these colours.
+
+This work follows [this RFC](https://github.com/alphagov/govuk_publishing_components/pull/287) to discuss the approach taken.
+
+## Adding to a component
+
+To add colours to a component, modify the component to follow the example below.
+
+```
+<%
+  brand ||= false
+  brand_helper = GovukPublishingComponents::Presenters::BrandHelper.new(brand)
+%>
+
+<div class="gem-c-component <%= brand_helper.get_brand %>">
+  <div class="gem-c-component__title <%= brand_helper.get_brand_element("border-color") %>">
```

Comment:
I think it's worth it creating separate method for this and `color` to avoid typos.

We mostly [avoid prefixing getters with `get_`](https://github.com/bbatsov/ruby-style-guide#accessor_mutator_method_names) (I'm not sure why Rubocop hasn't picked that up, I'll investigate).

Perhaps something like:

```diff
- <%= brand_helper.get_brand_element("border-color") %>">
+ <%= brand_helper.border_color_class %>
```

```diff
- <%= brand_helper.get_brand_element("color") %>">
+ <%= brand_helper.color_class %>
```